src/throttler/throttler.ts

```diff
@@ -31,7 +31,13 @@ export interface ThrottlerStats {
   elapsed: number;
 }
 
-export abstract class Throttler<T> {
+interface TaskData<T, R> {
+  task: T;
+  retryCount: number;
+  wait?: { resolve: (R: any) => void; reject: (err: Error) => void };
+}
+
+export abstract class Throttler<T, R> {
```

Comment:
last thing. Add a jsdoc block and describe what `T` and `R` are. It'll help people who haven't been picking it apart (like you and me) to understand how each is used.

Reply:
Sure thing.